<a href="https://colab.research.google.com/github/michaelherediaperez/mecanica_de_solidos_un/blob/main/codigos/cap_04/04_08_03_ejemplo_TP.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 4.8.3. Relación entre los esfuerzos principales obtenidos en el análisis bidimensional y tridimensional 

|Quién | Fecha | Qué hizo |
| ---  | ---   | ---      |
|Michael Heredia Pérez, <mherediap@unal.edu.co> | Julio 20, 2022 | Conversión de código de MATLAB a Python|

## Ejemplo de *tensión plana*


Al igual que como hicimos con sympy, podemos importar completamente la librería de numpy para trabajar con ella, sin embargo, recuerden que no es la práctica recomendada.

In [1]:
from numpy import *
set_printoptions(precision=4)   # imprime todo con 4 cifras decimales max

Si suponemos que este punto está sujeto a un estado de **tensión plana**, adicionalmente a $\sigma_x = $ − 1 Pa, $\sigma_y = $ = 2 Pa y $\tau_{xy} = $ − 3 Pa, se debe tener en cuenta que $\tay_{xz} = \tau_{yz} = \sigma_z = } 0 Pa. Armando la matriz (4.38) y resolviendo el problema de valores y vectores propios, obtenemos:


In [2]:
# Defino las variables.
sx  = -1
sy  = 2
txy = -3

# La matriz de esfuerzos.
sigma = array([[sx, txy, 0],
               [txy, sy, 0],
               [0,    0, 0]])
sigma

array([[-1, -3,  0],
       [-3,  2,  0],
       [ 0,  0,  0]])

In [3]:
# Calculo lo vectores y valores propios
valp, vecp = linalg.eig(sigma)

In [4]:
valp

array([-2.8541,  3.8541,  0.    ])

In [5]:
vecp

array([[-0.8507,  0.5257,  0.    ],
       [-0.5257, -0.8507,  0.    ],
       [ 0.    ,  0.    ,  1.    ]])

Es necesario enumerar de nuevo los valores y vectores propios de modo que $\sigma_1 \geq \sigma_2 \geq \sigma_3$,  y garantizar que el sentido escogido de los vectores propios cumpla la
regla de la mano derecha ($\mathbf{ \hat{n}_3 = \hat{n}_1 \times \hat{n}_2 }$), por lo que la solución al ejercicio se debe dar como:

$\sigma_3 , \sigma_2 , \sigma_1 =$

In [6]:
sort(valp)

array([-2.8541,  0.    ,  3.8541])

$\hat{n}_1 = $

In [7]:
vecp[:,1]

array([ 0.5257, -0.8507,  0.    ])

$\hat{n}_2 = $

In [8]:
vecp[:,2]

array([0., 0., 1.])

$\hat{n}_3 = $

In [9]:
vecp[:,0]

array([-0.8507, -0.5257,  0.    ])

ya que al hacer el producto cruz, obtenemos

In [10]:
n1 = vecp[:, 1]
n2 = vecp[:, 2]
cross(n1, n2)

array([-0.8507, -0.5257,  0.    ])

con lo cual cumplimos la regla de la mano derecha, al obtener el sentido requerido para $\hat{n}_3$.

Observe la discrepancia de signo entre los vectores propios $[− 0.8507, − 0.5257, 0]^\intercal$ y $[0.8507, 0.5257, 0]^\intercal$; esta diferencia no nos debe importar, ya que lo único relevante en el cálculo de los vectores propios es la dirección del vector mismo, no su sentido. De esta forma, si hubiéramos escogido el vector $\hat{n}_3$ como $[− 0.8507, − 0.5257, 0]^\intercal$, el sentido del vector $\hat{n}_2$ igualmente hubiera cambiado de modo que $\hat{n}_2$ hubiera sido $[0, 0, − 1]^\intercal$ con el fin de cumplir la regla de la mano derecha en la base formada por las direcciones principales.

Tenemos pues que dos de los valores propios corresponden a $(\sigma_1)_{xy}$  y $(\sigma_2)_{xy}$, mientras que el otro corresponde a $0$. Los vectores propios asociados son los mismos calculados en el caso bidimensional para los dos primeros, agregando un $0$ en la componente $z$, mientras que el vector propio asociado a 0 es [0,0, ± 1]T , siendo necesario ajustar su sentido (ver Figura 4.18).

Al igual que en el caso de deformación plana, debemos determinar la magnitud y los planos sobre los que actúa el esfuerzo cortante máximo.

En este ejemplo en particular,  estos son exactamente los mismos que los calculados en el ejemplo de deformación plana. Sin embargo, estos pueden variar. Supongamos que ambos $\sigma_1$ y $\sigma_2$ son cantidades positivas, entonces en este caso, tendremos un $\sigma_3 = 0$. Observe, por lo tanto, que en este caso cambiarían los valores de $\tau_{\max}$ y los vectores $\frac{ \hat{n}_1 - \hat{n}_3}{||\hat{n}_1 - \hat{n}_3||}$ y $\frac{ \hat{n}_1 + \hat{n}_3}{||\hat{n}_1 + \hat{n}_3||}$ ya no estarían sobre el plano $xy$.

En resumen, para el aso de tensión plana, se tiene que:

$$\sigma_1 = \max\left( (\sigma_1)_{xy}, (\sigma_1)_{xy}, 0 \right) $$

$$\sigma_2 = mediana\left( (\sigma_1)_{xy}, (\sigma_1)_{xy}, 0 \right) $$

$$\sigma_3 = \min\left( (\sigma_1)_{xy}, (\sigma_1)_{xy}, 0 \right) $$

$$ \tau_{\max} = \max\left( \frac{|(\sigma_1)_{xy}| }{2},  \frac{|(\sigma_2)_{xy}| }{2},  \frac{ |(\sigma_1)_{xy} - (\sigma_2)_{xy}| }{2}   \right)  $$ 